### ISSUE TITLE
rclone encrypted ACD -> local - not finding local files even though they exist

### ISSUE THREAD

**Issue by mathiasringhof** [rclone/rclone#1406]
I finished copying my file collection to Amazon Drive in combination with an encrypted remote. When checking if all files were uploaded rclone reports several local files as missing, e.g:
`2017/05/13 16:10:02 ERROR : Fullmetal Alchemist - The Sacred Star of Milos (2011)/fanart.jpg: File not in Local file system at /home/mathias/movies`

/home/mathias/movies is a CIFS mount.

They do exist however:
```
ll /home/mathias/movies/Fullmetal\ Alchemist\ -\ The\ Sacred\ Star\ of\ Milos\ \(2011\)/fanart.jpg
-rw-rw-rw- 1 mathias mathias 177K May 12 01:49 /home/mathias/movies/Fullmetal Alchemist - The Sacred Star of Milos (2011)/fanart.jpg
```

Here is the full output of **rclone check movies amzcrypt:movies -v**:
```
2017/05/13 16:16:24 INFO  : Encrypted amazon drive root 'crypt/foo123': Modify window not supported
2017/05/13 16:16:24 INFO  : Encrypted amazon drive root 'crypt/foo123': Building file list
2017/05/13 16:16:24 INFO  : Local file system at /home/mathias/movies: Building file list
2017/05/13 16:16:32 NOTICE: Local file system at /home/mathias/movies: 0 files not in Encrypted amazon drive root 'crypt/foo123'
2017/05/13 16:16:32 NOTICE: Encrypted amazon drive root 'crypt/foo123': 12 files not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Fullmetal Alchemist - The Sacred Star of Milos (2011)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Fullmetal Alchemist - The Sacred Star of Milos (2011)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Fullmetal Alchemist - The Sacred Star of Milos (2011)/Fullmetal Alchemist - The Sacred Star of Milos (2011) h264-720p AC3-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Harry Potter and the Goblet of Fire (2005)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Fullmetal Alchemist - The Sacred Star of Milos (2011)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Harry Potter and the Goblet of Fire (2005)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Harry Potter and the Goblet of Fire (2005)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Harry Potter and the Goblet of Fire (2005)/Harry Potter and the Goblet of Fire (2005) h264-1080p DTS-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Fullmetal Alchemist - The Sacred Star of Milos (2011)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 ERROR : Fullmetal Alchemist - The Sacred Star of Milos (2011)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:16:32 INFO  : Local file system at /home/mathias/movies: Waiting for checks to finish
2017/05/13 16:16:32 NOTICE: Local file system at /home/mathias/movies: 12 differences found
2017/05/13 16:16:32 NOTICE: Local file system at /home/mathias/movies: 1376 hashes could not be checked
2017/05/13 16:16:32 Failed to check: 12 differences found
```

I thought this might be an issue with the long path and changed the movie name to "Fullmetall Alchemist" both on the local mount as well as on the encrypted Amazon Drive:
`mv Fullmetal\ Alchemist\ -\ The\ Sacred\ Star\ of\ Milos\ \(2011\) Fullmetal\ Alchemist`

Now a different set of files is is listed as missing (a movie that wasn't listed before), but again exactly 12 differences:
```
2017/05/13 16:22:20 INFO  : Encrypted amazon drive root 'crypt/foo123': Modify window not supported
2017/05/13 16:22:20 INFO  : Encrypted amazon drive root 'crypt/foo123': Building file list
2017/05/13 16:22:20 INFO  : Local file system at /home/mathias/movies: Building file list
2017/05/13 16:22:31 NOTICE: Local file system at /home/mathias/movies: 0 files not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/13 16:22:31 NOTICE: Encrypted amazon drive root 'crypt/foo123': 12 files not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Harry Potter and the Goblet of Fire (2005)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Harry Potter and the Goblet of Fire (2005)/Harry Potter and the Goblet of Fire (2005) h264-1080p DTS-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Harry Potter and the Goblet of Fire (2005)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005) h264-720p AC3-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Harry Potter and the Goblet of Fire (2005)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 16:22:31 INFO  : Local file system at /home/mathias/movies: Waiting for checks to finish
2017/05/13 16:22:31 NOTICE: Local file system at /home/mathias/movies: 12 differences found
2017/05/13 16:22:31 NOTICE: Local file system at /home/mathias/movies: 1376 hashes could not be checked
2017/05/13 16:22:31 Failed to check: 12 differences found
```

I tried reducing the number of checkers to 1 with the same result.

Please let me know if I can supply anything else to help figure this out. 

EDIT: I removed Ghost in the shell and let clone copy it down. Same result. Interestingly, renaming Fullmetal Alchemist back I still have the output from above (Ghost + Harry P not in local filesystem).

> What is your rclone version (eg output from `rclone -V`)
1.3.6, but same issue with latest beta

> Which OS you are using and how many bits (eg Windows 7, 64 bit)
CentOS 7
Kernel 3.10.0-514.16.1.el7.x86_64

> Which cloud storage system are you using? (eg Google Drive)
Amazon Drive + Encrypted

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
./rclone check ~/movies amzcrypt:movies

**Comment by mathiasringhof** [comment]
Since right now `rclone check amzcrypt:movies movies -v" returns *Harry Potter* and *Ghost in the Shell* as not being available locally I'm using Harry Potter.

So in full:
`rclone check amzcrypt:movies movies -v` returns:
```
2017/05/13 17:41:07 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': Modify window not supported
2017/05/13 17:41:07 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': Building file list
2017/05/13 17:41:07 INFO  : Local file system at /home/mathias/movies: Building file list
2017/05/13 17:41:16 NOTICE: Local file system at /home/mathias/movies: 5 files not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/13 17:41:16 ERROR : fanart.jpg: File not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/13 17:41:16 ERROR : movie.nfo: File not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/13 17:41:16 ERROR : poster.jpg: File not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/13 17:41:16 ERROR : movieset-fanart.jpg: File not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/13 17:41:16 ERROR : movieset-poster.jpg: File not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/13 17:41:16 NOTICE: Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': 12 files not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Harry Potter and the Goblet of Fire (2005)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Harry Potter and the Goblet of Fire (2005)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Harry Potter and the Goblet of Fire (2005)/Harry Potter and the Goblet of Fire (2005) h264-1080p DTS-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005) h264-720p AC3-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Harry Potter and the Goblet of Fire (2005)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/13 17:41:16 INFO  : Local file system at /home/mathias/movies: Waiting for checks to finish
2017/05/13 17:41:16 NOTICE: Local file system at /home/mathias/movies: 17 differences found
2017/05/13 17:41:16 NOTICE: Local file system at /home/mathias/movies: 1376 hashes could not be checked
2017/05/13 17:41:16 Failed to check: 17 differences found
```

`rclone check amzcrypt:"movies/Harry Potter and the Goblet of Fire (2005)" movies/Harry\ Potter\ and\ the\ Goblet\ of\ Fire\ \(2005\) -v` returns no differences:
```
2017/05/13 17:38:56 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4/fgha5hfu12bh81bpketfle02qi3ku8tum470bh4l3p6lc3ehvv0gjo1olt9eq4h8hp2hoarinqbp2': Modify window not supported
2017/05/13 17:38:56 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4/fgha5hfu12bh81bpketfle02qi3ku8tum470bh4l3p6lc3ehvv0gjo1olt9eq4h8hp2hoarinqbp2': Building file list
2017/05/13 17:38:56 INFO  : Local file system at /home/mathias/movies/Harry Potter and the Goblet of Fire (2005): Building file list
2017/05/13 17:38:56 NOTICE: Local file system at /home/mathias/movies/Harry Potter and the Goblet of Fire (2005): 0 files not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4/fgha5hfu12bh81bpketfle02qi3ku8tum470bh4l3p6lc3ehvv0gjo1olt9eq4h8hp2hoarinqbp2'
2017/05/13 17:38:56 NOTICE: Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4/fgha5hfu12bh81bpketfle02qi3ku8tum470bh4l3p6lc3ehvv0gjo1olt9eq4h8hp2hoarinqbp2': 0 files not in Local file system at /home/mathias/movies/Harry Potter and the Goblet of Fire (2005)
2017/05/13 17:38:56 INFO  : Local file system at /home/mathias/movies/Harry Potter and the Goblet of Fire (2005): Waiting for checks to finish
2017/05/13 17:38:56 NOTICE: Local file system at /home/mathias/movies/Harry Potter and the Goblet of Fire (2005): 0 differences found
2017/05/13 17:38:56 NOTICE: Local file system at /home/mathias/movies/Harry Potter and the Goblet of Fire (2005): 6 hashes could not be checked
```

Here's the output of the two `clone lsl` commands:
```
➜  ~ rclone lsl movies/Harry\ Potter\ and\ the\ Goblet\ of\ Fire\ \(2005\)
13229470679 2017-05-10 07:56:55.682000000 Harry Potter and the Goblet of Fire (2005) h264-1080p DTS-6ch.mkv
    15010 2017-05-10 07:46:41.283000000 movie.nfo
    81968 2017-05-10 07:46:42.066000000 movieset-poster.jpg
   679788 2017-05-10 07:46:41.336000000 fanart.jpg
   369346 2017-05-10 07:46:42.794000000 movieset-fanart.jpg
   154463 2017-05-10 07:46:41.956000000 poster.jpg
➜  ~ rclone lsl amzcrypt:"movies/Harry Potter and the Goblet of Fire (2005)"
   154463 2017-05-10 07:46:41.956000000 poster.jpg
    81968 2017-05-10 07:46:42.066000000 movieset-poster.jpg
   369346 2017-05-10 07:46:42.794000000 movieset-fanart.jpg
    15010 2017-05-10 07:46:41.283000000 movie.nfo
   679788 2017-05-10 07:46:41.336000000 fanart.jpg
13229470679 2017-05-10 07:56:55.682000000 Harry Potter and the Goblet of Fire (2005) h264-1080p DTS-6ch.mkv
```

Thanks for the swift reply!

**Comment by mathiasringhof** [comment]
I wanted to do some digging in the code and noticed that rclone 1.36 using MacOS works just fine...

```
➜  ~ rclone check amzcrypt:movies movies -v
2017/05/17 15:58:41 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': Modify window not supported
2017/05/17 15:58:41 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': Building file list
2017/05/17 15:58:41 INFO  : Local file system at /home/mathias/movies: Building file list
2017/05/17 15:58:49 NOTICE: Local file system at /home/mathias/movies: 0 files not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/17 15:58:49 NOTICE: Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': 12 files not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Harry Potter and the Goblet of Fire (2005)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/fanart.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Harry Potter and the Goblet of Fire (2005)/Harry Potter and the Goblet of Fire (2005) h264-1080p DTS-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Harry Potter and the Goblet of Fire (2005)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Harry Potter and the Goblet of Fire (2005)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Harry Potter and the Goblet of Fire (2005)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movie.nfo: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005) h264-720p AC3-6ch.mkv: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 ERROR : Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/poster.jpg: File not in Local file system at /home/mathias/movies
2017/05/17 15:58:49 INFO  : Local file system at /home/mathias/movies: Waiting for checks to finish
2017/05/17 15:58:49 NOTICE: Local file system at /home/mathias/movies: 12 differences found
2017/05/17 15:58:49 NOTICE: Local file system at /home/mathias/movies: 1376 hashes could not be checked
2017/05/17 15:58:49 Failed to check: 12 differences found```

macOS:
```
➜  Movies rclone check amzcrypt:movies /Volumes/Movies -v
2017/05/17 15:58:52 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': Modify window not supported
2017/05/17 15:58:52 INFO  : Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': Building file list
2017/05/17 15:58:52 INFO  : Local file system at /Volumes/Movies: Building file list
2017/05/17 15:59:01 NOTICE: Local file system at /Volumes/Movies: 0 files not in Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4'
2017/05/17 15:59:01 NOTICE: Encrypted amazon drive root 'crypt/an8cshp5kb3h1kbs0obmle96e4': 0 files not in Local file system at /Volumes/Movies
2017/05/17 15:59:01 INFO  : Local file system at /Volumes/Movies: Waiting for checks to finish
2017/05/17 15:59:01 NOTICE: Local file system at /Volumes/Movies: 0 differences found
2017/05/17 15:59:01 NOTICE: Local file system at /Volumes/Movies: 1388 hashes could not be checked
```

The folders are the same from an OS perspective, although I had to use Ignore Case in sort so I get the same sorting result.

Linux:
```
➜  movies find . | sort -f | md5sum
9c88a69a7441145ee429cd2074c0907e  -
```

macOS:
```
➜  Movies find . | sort -f | md5
9c88a69a7441145ee429cd2074c0907e
```

One of the directories reported as missing for reference.

Linux:
```
➜  movies find . | grep Individual
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-poster.jpg
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-fanart.jpg
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005) h264-720p AC3-6ch.mkv
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movie.nfo
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/fanart.jpg
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/poster.jpg
➜  movies find . | grep Individual | md5sum
a5a5b6ade057ddc6588f8b6ffc9c13d1  -
```

macOS:
```
➜  Movies find . | grep Individual
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-poster.jpg
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movieset-fanart.jpg
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005) h264-720p AC3-6ch.mkv
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/movie.nfo
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/fanart.jpg
./Ghost in the Shell - Stand Alone Complex - Individual Eleven (2005)/poster.jpg
➜  Movies find . | grep Individual | md5
a5a5b6ade057ddc6588f8b6ffc9c13d1
```

Not sure what I'm missing here, doesn't make sense to me right now (which usually indicates user error....).

**Comment by mathiasringhof** [comment]
Due to switching from CentOS 7.3 to Debian 9 I found that the problem disappeared for me. I used the exact same configuration in fstab for the CIFS mounts so I assume some other setting influenced the results. I switched remote to Google before switching server so I'm 99% sure the error is caused by CentOS (or my particular CentOS install, although I didn't do much except install a few pieces of software).
